system-tests/test/component_testing_spec.ts

```diff
@@ -122,6 +122,14 @@ describe(`Angular CLI major versions`, () => {
       browser: 'chrome',
       expectedExitCode: 0,
     })
+
+    systemTests.it(`v${majorVersion} Zone.JS does not break Mocha`, {
+      project: `angular-${majorVersion}`,
+      spec: 'src/app/zonejs-mocha-skip.cy.ts,src/app/zonejs-mocha-only.cy.ts',
+      testingType: 'component',
```

Comment:
I actually think this is the wrong place for this test. Also, we are not actually verifying that `it.skip` actually does what's it supposed to, only that it is **available**.

We could move this to a Cypress in Cypress test in `app/cypress/e2e`, and we could then assert against the command log that the correct number of tests are executed and/or skipped? That would verify not only that it works, but `skip` is the correct one (since we will see it visually marked as skipped in the Commanad Log, as well as the number of executed tests).

Reply:
You would add it here: https://github.com/cypress-io/cypress/blob/develop/packages/app/cypress/e2e/cypress-in-cypress-component.cy.ts

If you haven't seen the Cypress in Cypress yet it's really cool - you can still use this project (in system tests) but you can now assert things are actually happening in the UI, too.

What do you think?

Reply:
It looks like `cypress-in-cypress` is configured to use React for CT, so were you thinking we would stand up a new `cypress-in-cypress` permutation using Angular?

Reply:
Took the approach Zach proposed to write some tests to validate the behavior of `skip` and `only`. Definitely not as elegant as `cypress-in-cypress`, but very performant and seems to provide the validation that was requested

Reply:
Oh I hadn't seen that @ZachJW34, neat.

The reason I suggested cypress-in-cypress is we have these really cool helpers: https://github.com/cypress-io/cypress/blob/6ee305ba411c646298018a965d249a418ac68671/packages/app/cypress/e2e/support/execute-spec.ts#L14-L18.

You can do

```js
cy.waitForSpecToFinish({
  passCount: 4
  failCount: 0
  pendingCount: 1
})
```

Which is exactly what we want here - maybe we should find a way to share more of these commands between cypress-in-cypress and system-tests in the future.